pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -211,6 +212,34 @@ func (bdl *BlockDeviceList) Filter(predicateKeys ...string) *BlockDeviceList {
 	return filteredBlockDeviceList
 }
 
+// GetUsableBlockDevices returns custom blockdevice list which can be used
+func (bdl *BlockDeviceList) GetUsableBlockDevices(spcName, namespace string) (*BlockDeviceList, error) {
+	// Initialize filtered block device list
+	if bdl == nil {
+		return nil, errors.Errorf("failed to get usable blockdevices: blockdevicelist is nil")
+	}
+	filteredBlockDeviceList := &BlockDeviceList{
+		BlockDeviceList: &ndm.BlockDeviceList{},
+		errs:            nil,
+	}
+	bdcClient := bdc_v1alpha1.NewKubeClient().WithNamespace(namespace)
+	for _, bdObj := range bdl.Items {
+		if bdObj.Status.ClaimState == ndm.BlockDeviceClaimed {
+			bdcName := bdObj.Spec.ClaimRef.Name
+			bdcObj, err := bdcClient.Get(bdcName, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			if bdcObj.Labels[string(apis.StoragePoolClaimCPK)] == spcName {
+				filteredBlockDeviceList.Items = append(filteredBlockDeviceList.Items, bdObj)
+			}
+		} else if bdObj.Status.ClaimState == ndm.BlockDeviceUnclaimed {
+			filteredBlockDeviceList.Items = append(filteredBlockDeviceList.Items, bdObj)
+		}
```

Comment:
The repetition in `else if`  block can be avoided.

Reply:
If we avoid it how can we get unclaimed blockdevices?